### PR TITLE
Add missing urls for hash modules

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik

### CODE REVIEW

**Comment by steveklabnik** [comment]
@bors: delegate=kmcallister


ugh i should learn how to use bors

File: src/libcore/hash/mod.rs [hunk 1]
```diff
@@ -127,6 +129,11 @@ mod sip;
 ///     }
 /// }
 /// ```
+///
+/// [`Eq`]: cmp/trait.Eq.html
+/// [`HashMap`]: ../collections/struct.HashMap.html
+/// [`HashSet`]: ../collections/struct.HashSet.html
```
Comment (kmcallister) [inline src/libcore/hash/mod.rs]: I think these links will be broken on the `libcore` docs.

File: src/libcore/hash/mod.rs [hunk 2]
```diff
@@ -151,11 +158,11 @@ pub trait Hasher {
     #[stable(feature = "rust1", since = "1.0.0")]
     fn finish(&self) -> u64;
 
-    /// Writes some data into this `Hasher`
+    /// Writes some data into this `Hasher`.
     #[stable(feature = "rust1", since = "1.0.0")]
     fn write(&mut self, bytes: &[u8]);
 
-    /// Write a single `u8` into this hasher
+    /// Write a single `u8` into this hasher.
```
Comment (kmcallister) [inline src/libcore/hash/mod.rs]: Can you fix the tense mismatch between `Writes` above and `Write` here?

**Comment by GuillaumeGomez** [comment]
Updated (waiting for url check...).

**Comment by frewsxcv** [comment]
Oh just kidding. The delegate was for @kmcallister. Will let them r+.

**Pull request merged**